askama_shared/src/generator.rs

```diff
@@ -353,9 +353,11 @@ impl<'a> Generator<'a> {
 
     fn write_expr(&mut self, ws: &WS, s: &Expr) {
         self.handle_ws(ws);
-        self.write("writer.write_fmt(format_args!(\"{}\", ");
+        self.write("let askama_expr = &");
         self.visit_expr(s);
-        self.writeln("))?;");
+        self.writeln(";");
+        self.writeln("writer.write_fmt(format_args!(\"{}\", \
+                      &::askama::MarkupDisplay::from(askama_expr)))?;");
```

Comment:
same question here - does this need to be referenced? I think `format_args!` will figure it out for you anyway.

Reply:
I've typically found that the formatting functions only want references; they don't take ownership. This makes sense to me, so it would be good to make that clear in the code.